oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.remote_attestation.v1;
+
+option java_multiple_files = true;
+option java_package = "com.google.oak.remote_attestation.v1";
+
+// This proto defines the layered DICE Attestation Evidence.
+//
+// DICE provides a mechanism for combining software measurements and corresponding certificates into
+// a chain, where each element (called “layer”) represents a piece of software loaded into the VMs
+// memory. These layers are loaded sequentially, i.e. the previous layer loads the next layer. The
+// previous layer is also responsible for measuring the next layer, generating a private key and a
+// certificate for it.
+//
+// <https://trustedcomputinggroup.org/wp-content/uploads/TCG_DICE_Attestation_Architecture_r22_02dec2020.pdf>
+// <https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf>
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  AMD_SEV_SNP = 1;
+  INTEL_TDX = 2;
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
+  // CBOR/COSE/CWT encryption public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes encryption_public_key_certificate = 0;
+  // CBOR/COSE/CWT signing public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes signing_public_key_certificate = 1;
+}
+
+message Evidence {
+  // Layer0 attestation evidence.
+  RootLayerEvidence root_layer_evidence = 0;
+  // Layer1..LayerN-1 attestation evidence.
+  repeated LayerEvidence layer_evidence = 1;
+  // Final layer keys.
+  optional ApplicationKeys application_keys = 2;
+}
+
+// Endorsement contains statements that some entity (e.g., a hardware provider) vouches for the
+// integrity of claims about the TEE or the software running on it.
+// The name is chosen to match the RATS terminology:
+// <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
+message Endorsement {
+  // The serialized TEE certificate(s). The details of the format and how the certificate(s) are
+  // encoded into this byte array are implementation-specific. In case of AMD-SEV-SNP, as described
+  // in https://www.amd.com/system/files/TechDocs/57230.pdf, there are three different certificates
+  // packaged in two different files.
+  bytes tee_certificates = 1;
+
+  // Transparent Release endorsement.
+  TransparentReleaseEndorsement transparent_release_endorsement = 2;
```

Comment:
I would assume there would be multiple endorsements here, typically one for each component (e.g. Stage 0, Initial RAM disk etc.)

@rbehjati is this assumption accurate, or will all the components be rolled into a single endorsement?

Reply:
Yes. There should be multiple of these.

Reply:
I'm surprised to find this in the DICE context. Isn't it orthogonal? Maybe it should be a different file / package. If we keep them together, it means that if we want to change the format of the evidence that the launcher returns to the client, we need to modify this file, which also causes stage0 to have to be rebuilt. I think those should be kept distinct, since they are used by completely different systems.